.github/workflows/ci.yml

```diff
@@ -0,0 +1,276 @@
+---
+name: Continuous Integration tests
+
+on: [pull_request]
+
+jobs:
+  main_tests:
+    name: Main tests
+    runs-on: ${{ matrix.os }}
+    strategy:
+      # Duplicate changes to this matrix to 'poc_tests'
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: ['3.9']
+        # Possible values: ['2.7', '3.6', '3.7', '3.8', '3.9', '3.10.0-rc.2']
+        include:
+          - os: ubuntu-latest
+            python-version: '3.6'
+          - os: ubuntu-latest
+            python-version: '3.7'
+          - os: ubuntu-latest
+            python-version: '3.8'
+          ## Expected failure
+          - os: ubuntu-latest
+            python-version: '3.10.0-rc.2'
+
+    steps:
+      - uses: actions/checkout@v2
+
+      # - template: azure-templates/ccache.yml
+      #   parameters:
+      #     pythonVersion: $(python.version)
+      # - template: azure-templates/python.yml
+      #   parameters:
+      #     pythonVersion: $(python.version)
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install/Upgrade Python dependencies
+        run: python -m pip install --upgrade pip wheel
+
+      - name: Build
+        run: python -m pip install .
+
+      - name: Run tests
+        run: |
+          python -m pip install pytest pytest-xdist
+          python -m pytest -n auto test/
+
+
+
+  poc_tests:
+    name: Proof of concept tests
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: ['3.9']
+        include:
+          - os: ubuntu-latest
+            python-version: '3.6'
+          - os: ubuntu-latest
+            python-version: '3.7'
+          - os: ubuntu-latest
+            python-version: '3.8'
+          - os: ubuntu-latest
+            python-version: '3.10.0-rc.2'
```

Comment:
Never mind this does fail at runtime https://github.com/dralley/hpy/pull/1/checks?check_run_id=3551010371#step:5:214